tests/common/osutil/test_default.py

```diff
@@ -88,14 +88,22 @@ def test_mount_dvd_failure(self, _):
                         self.assertTrue(msg in ustr(ose))
                         self.assertTrue(patch_run.call_count == 6)
 
-    def test_get_first_if(self):
+    @patch('azurelinuxagent.common.osutil.default.DefaultOSUtil.get_primary_interface', return_value='eth0')
+    @patch('azurelinuxagent.common.osutil.default.DefaultOSUtil._get_interface_list_as_string',
+           return_value=b'eth0\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02\x00\x00\x00\xac\x11\xba9\x00\x00\x00\x00\x00')
```

Comment:
I question the \xba9 value in this b'' string. Everything else is either an ordinary character or a two-digit hex value (8 bits); this one 16-bit hex value (without leading zero) makes it hard to check the string. (I know \xc1\x11 are the first two octets of the IPv4 address, but I can't _see_ the last two octets as individual octets.)

Reply:
It looks like it is just a typo that should be `\xba\x39`

Reply:
I copied the value from the debugger (honest), and the unit test asserts the correct IP address.

Reply:
Doh! @uiri is correct; the '9' is an absolute character, not part of the previous 8-bit escape. I'd like to see '9' replaced by \x39 to make it clear that it represents an octet in an IP address and not a printable character, but that is a stylistic nit.